source/blog/042-wide-gamut-colors-in-sass.md

```diff
@@ -0,0 +1,279 @@
+
+- We can inspect the space of a color using `color.space($color)`
+- We can ask if the color is in a legacy space with `color.is-legacy($color)`
+- We can *convert* a color from one space to another using `color.to-space($color, $space)`
+
+All of these functions are provided by the built-in [Sass Color Module](https://sass-lang.com/documentation/modules/color/):
+
+```scss
+@use 'sass:color';
+$brand: MediumVioletRed;
+
+// results: rgb, true
+$initial: color.space($brand);
+$is-legacy: color.is-legacy($brand);
+
+// result: oklch(55.34% 0.2217 349.7)
+$converted: color.to-space($brand, 'oklch');
+
+// results: oklch, false
+$new-space: color.space($brand);
+$is-legacy: color.is-legacy($brand);
+```
+
+Once we can convert a color between spaces, we no longer consider those colors to be *equal*. But we can ask if they would render as 'the same' color, using the `color.same()` function:
+
+```scss
+@use 'sass:color';
+$orange-rgb: #ff5f00;
+$orange-oklch: oklch(68.72% 20.966858279% 41.4189852913deg);
+
+// result: false
+$equal: $orange-rgb == $orange-oklch;
+
+// result: true
+$same: color.same($orange-rgb, $orange-oklch);
+```
+
+We can inspect the individual channels of a color using `color.channel()`. By default, it only supports channels that are available in the color's own space, but we can pass the `$space` parameter to return the value of the channel value after converting to the given space:
+
+```scss
+@use 'sass:color';
+$brand: hsl(0 100% 25.1%);
+
+// result: 25.1%
+$hsl-lightness: color.channel($brand, "lightness");
+
+// result: 37.67%
+$oklch-lightness: color.channel($brand, "lightness", $space: oklch);
+```
+
+CSS has also introduced the concept of 'powerless' and 'missing' color channels. For example, an `hsl` color with `100%` lightness will *always be white*. In that case, we can consider both the `hue` and `saturation` channels to be powerless. Changing their value won't have any impact on the resulting color. Sass allows us to ask if a channel is powerless using the `color.is-powerless()` function:
+
+```scss
+@use 'sass:color';
+$grey: hsl(0 0% 60%);
+
+// result: true, because saturation is 0
+$hue-powerless: color.is-powerless($grey, "hue");
+
+// result: false
+$hue-powerless: color.is-powerless($grey, "lightness");
+```
+
+Taking that a step farther, CSS also allows us to explicitly mark a channel as 'missing' or unknown. That can happen automatically if we convert a color like `gray` into a color space like `oklch` -- we don't have any information about the `hue`. We can also create colors with missing channels explicitly by using the `none` keyword, and inspect if a color channel is missing with the `color.is-missing()` function:
+
+```scss
+@use 'sass:color';
+$brand: hsl(none 100% 25.1%);
+
+// result: false
+$missing-lightness: color.is-missing($brand, "lightness");
+
+// result: true
+$missing-hue: color.is-missing($brand, "hue");
+```
+
```

Comment:
It's probably worth adding a note here that other than `is-missing()` and places where the CSS spec explicitly cares about missing channels, we treat them as though they had value 0.